### ISSUE TITLE
Named algorithm outputs in modeler fail when used as inputs

### ISSUE THREAD

**Issue by david-laverty** [alexbruy/processing-taudem#8]
There appears to be the issue that was reported in Bug report #19372 on the main QGIS modeller happening here as well when this plugin is used within the QGIS modeller.

I am trying to run the following model to create a catchment just as i have done in QGIS2.18 but if the 1st algorithm is given a name the model fails. If run with no names on 1st stage algorithms, the final output comes out of the model as expected. 

The issue arises as the 'D8 flow direction' algorithm output is necessary in my next model which allows a catchment to be created from a specific point once the Contributing areas output has been viewed to know where to put the Catchment point. 

Model_pt1
![image](https://user-images.githubusercontent.com/48278498/54749477-9236fe80-4bcc-11e9-8520-41bb38ceb33c.png)

Model pt2
![image](https://user-images.githubusercontent.com/48278498/54749839-acbda780-4bcd-11e9-8d16-64ce9f0e94b7.png)

If there is anything I can do to help resolve this issue let me know, I just did not get the full detail of how this was fixed from the Bug report.

**Comment by alexbruy** [comment]
This is not a plugin issue. If there is an issue in the Processing Modeler behavior it should be reported in the QGIS bugtracker.

**Comment by david-laverty** [comment]
Ok. Thanks. Sorry for always pestering you but I truely enjoy this plugin and am very grateful for you getting it to work on the new QGIS version.

I will check again tomorrow in the office to see if this issue has been resolved in the new 3.6.1 version

The only other issue I have is that there is not an iterator button on the plugin now as there was in the old 2.18 version. Do you believe this is also a QGIS issue or a plugin dependent one?

Kind regards

David

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by alexbruy** [comment]
> The only other issue I have is that there is not an iterator button on the plugin now as there was in the old 2.18 version. Do you believe this is also a QGIS issue or a plugin dependent one?

Do you mean an iterate buttons near vector parameters? If this is useful I can bring them back, but from my experience such functionality was not used widely with TauDEM.

**Comment by david-laverty** [comment]
yes, thats the one. I found it very useful for calculating multiple catchments at once from further downstream each time using the D8 Contributing area. This allowed you to see how a catchment grew as you progressed downstream without running each point individually.

If you have time and its not to much trouble I would appreciate that feature being added back into the plugin.

Also, is there a way to allow layers with spaces in their names to be used again for processing. The only way now I can get any process to run is to make sure the layer name and also the full path have no spaces in their name.  This is quite time consuming at times as most of the LiDAR we have here at work is already saved in folders with name spaces. But this is a problem I can live with if it is the too much hassle to fix, Was just another thing I found had been removed from the QGIS 3 version compared to 2.18

Again thanks for the consideration as I'm sure you are a very busy person.

Kind Regards,

David Laverty
________________________________
